cli/azd/pkg/async/task_context.go

```diff
@@ -0,0 +1,97 @@
+//nolint:all
+package async
+
+// Task function definition
+type TaskRunFunc[R comparable] func(ctx *TaskContext[R])
```

Comment:
We are getting different types of contexts around the code.  We have the go-context (context.Context), the azdContext and now the taskContext.  Let's start being more descriptive on the name of the variables to know what type of context we are looking at

```suggestion
type TaskRunFunc[R comparable] func(taskContext *TaskContext[R])
```